backend/internal/handlers/login.go

```diff
@@ -0,0 +1,38 @@
+package handlers
+
+import (
+	"errors"
+	"log"
+	"net/http"
+)
+
+type loginRequestBody struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+var v http.HandlerFunc
+
+// Login handles user logging in
+func Login(w http.ResponseWriter, r *http.Request) {
+	var body loginRequestBody
+
+	err := decodeJSONBody(w, r, &body)
+	if err != nil {
+		var mr *malformedRequest
+		if errors.As(err, &mr) {
+			http.Error(w, mr.msg, mr.status)
+		} else {
+			log.Println(err.Error())
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		}
+		return
+	}
+
+	// TODO: pwd should be saved in db and bcrypt-ified
+	if body.Email == "admin@goteleport.com" && body.Password == "admin@goteleport.com" {
+		w.Write([]byte("Sign in succeeded"))
+	} else {
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+	}
+}
```

Comment:
If the login implementation is not complete, I would not include it. It's hard to review incomplete code.

Reply:
Hmm, yeah makes sense that it's hard to review incomplete code. My intention with this half-baked endpoint was to give me something to hit to make sure that all the other moving parts (SSL/Docker/serving-a-spa-with-go/browser-side-routing) were actually configured correctly.

My initial thinking was that adding the database and session management would be my next PR, though I could just tack that on to this one if you think that makes more sense. Alternately I could make it clear that this is just a toy by renaming it something like "fake_endpoint". Open to better ideas.

Reply:
That's fine, will review it in the next PR.